doc/deploy.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# Copyright 2013-2021 [The Foundry Visionmongers Ltd]
+# SPDX-License-Identifier: Apache-2.0
+
+# A script to deploy an existing documentation build in
+# the current working directory to the target branch of
+# the specifed local repository clone.
+#
+# The script expects the build to exist at `./html`,
+# and the target repo to have a writable `docs` branch.
+
+set -e
+
+#
+# Constants
+#
+
+# The target branch to be updated with the new build
+target_branch=docs
+# The path to store the docs under in the target branch
+html_dir=docs
+
+#
+# Args
+#
+
+script_name=$0
+repo_dir=$1
+
+usage()
+{
+    echo "usage: $script_name <target-repo-path>"
+}
+
+if [[ -z "${repo_dir}" ]]; then
+    echo "ERROR: No target repository path specified"
+    usage
+    exit 1
+fi
+
+#
+# Helpers
+#
+
+script_cwd=`pwd`
+function restore_cwd() {
+    cd "${script_cwd}"
+}
+
+#
+# Deployment
+#
+
+# Ensure the checkout has no uncommited changes so we
+# only deploy a commited version with an identifiable hash.
+if ! git diff --quiet HEAD
+then
+    echo "ERROR: Please commit all changes and rebuild documentation before deploying"
+    git status -s
+    exit 1
+fi
+
+commit_hash=`git rev-parse --short HEAD`
+target_dir="${repo_dir}/${html_dir}"
+
+echo "Copying local build (commit ${commit_hash}) to ${target_dir}"
+
+# Clean up any existing build in the target should files
+# have been removed in the new build.
+rm -rf "${target_dir}"
+# Copy in the new build
+cp -r ./html "${target_dir}"
+
+# See if there are any resulting changes in the target
+# branch, commit and push if so.
+
+# Ensure we restore the working dir if any actions in the
+# target repo fail
+trap restore_cwd EXIT
```

Comment:
[question] This should only be necessary if you're `source`ing the script, is that something we envision? Also perhaps `pushd`/`popd` is of interest.

Reply:
Doh! Of course yes. No not intending a `source`. Something was stuck in my mind that one needed to do such house keeping. But as you say, it is redundant. Removed [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/103/commits/a47f8d5cbf805d9450cb6ae5b7cb2c51eefa1b41).